### ISSUE TITLE
Missing invalid-paramspec for legacy ParamSpec used as PEP 695 ParamSpec default

### ISSUE THREAD

**Issue by denyszhak** [astral-sh/ty#2874]
**Summary:**

`ty` does not report `invalid-paramspec` when a legacy `ParamSpec` is used as the default for a PEP 695 `ParamSpec` type parameter. Currently, no diagnostic being reported.

```py
from typing import ParamSpec

P1 = ParamSpec("P1")

class C[**P = P1]:
    pass
```

**Expected:**

An invalid-paramspec diagnostic like: "Cannot combine legacy ParamSpec defaults with PEP 695 type parameters"

**Why this should error:**

Mixing traditional TypeVar/ParamSpec declarations with PEP 695 type parameters is disallowed by the typing spec:
https://typing.python.org/en/latest/spec/generics.html#compatibility-with-traditional-typevars

**Maybe related:**

ParamSpec validation issue: [#1861](https://github.com/astral-sh/ty/issues/1861)

**Comment by charliermarsh** [comment]
Looks like we omit this now.
